MWPW-164093: Adobe Home 'Pricing Widget' merch card

Axelcureno opened the pull request:
This PR introduces Adobe Home 'Pricing Widget' merch card with 2 variations:
<img width="515" alt="Screenshot 2025-02-07 at 11 24 32 AM" src="https://github.com/user-attachments/assets/001cdee4-1f53-4db6-a365-9a3407d749b0" />
<img width="511" alt="Screenshot 2025-02-07 at 11 24 38 AM" src="https://github.com/user-attachments/assets/e0a15dba-8a56-4939-a284-d91b4e65c6e9" />
<img width="512" alt="Screenshot 2025-02-07 at 11 24 46 AM" src="https://github.com/user-attachments/assets/5b252d64-6ef6-44a2-a3e6-a204af4dfb41" />

Resolves: [MWPW-164093](https://jira.corp.adobe.com/browse/MWPW-164093)

M@S Studio Test URL: https://mwpw-164093--mas--adobecom.hlx.page/studio.html?milolibs=MWPW-164093--milo--adobecom#path=adobe-home&query=pricing
Test page: https://mwpw-164093--milo--adobecom.hlx.live/libs/features/mas/docs/adobe-home.html?martech=off

Test URLs:

Before: https://main--milo--adobecom.hlx.live/docs/library/kitchen-sink/merch-card?martech=off
After: https://mwpw-164093--milo--adobecom.hlx.live/docs/library/kitchen-sink/merch-card?martech=off

yesil commented on libs/features/mas/docs/checkout-button.html:
I think this is not in the navigation menu(lost in previous PRs) can you add it back?

yesil commented on libs/features/mas/src/hydrate.js:
```diff
@@ -175,16 +285,40 @@ function createSpectrumSwcButton(cta, aemFragmentMapping, isOutline, variant) {
+        try {
+          await customElements.whenDefined('checkout-button');
+          const CheckoutButtonEl = customElements.get('checkout-button');
+          const checkoutBtn = CheckoutButtonEl.createCheckoutButton({}, cta.innerHTML);
```
I think this block causes runtime errors in ah gallery:

mas.js:2370 Failed to initialize checkout-button logic: TypeError: Cannot read properties of null (reading 'setAttribute')
    at mas.js:2370:30403

yesil commented on libs/features/mas/src/hydrate.js:
```diff
@@ -175,16 +285,40 @@ function createSpectrumSwcButton(cta, aemFragmentMapping, isOutline, variant) {
+          const checkoutBtn = CheckoutButtonEl.createCheckoutButton({}, cta.innerHTML);
+    
+          for (const attr of cta.attributes) {
+            checkoutBtn.setAttribute(attr.name, attr.value);
```
I wonder why this is needed?

yesil commented on libs/features/mas/src/hydrate.js:
```diff
@@ -175,16 +285,40 @@ function createSpectrumSwcButton(cta, aemFragmentMapping, isOutline, variant) {
-            /* c8 ignore next 3 */
+    (async () => {
+        try {
+          await customElements.whenDefined('checkout-button');
```
I think it is safe to assume that `checkout-button` is already defined before we hit here, if not wee need to check how mas.js is bundled.
I would avoid adding checks for web components that we know are in the bundle.

yesil commented on libs/features/mas/src/hydrate.js:
```diff
@@ -175,16 +285,40 @@ function createSpectrumSwcButton(cta, aemFragmentMapping, isOutline, variant) {
+          spectrumCta.addEventListener('click', (e) => {
             e.stopPropagation();
-            cta.click();
+            if (typeof checkoutBtn.clickHandler === 'function') {
```
isn't `checkoutBtn.click()` not enough?

Axelcureno replied:
yes, fixed

yesil commented on libs/features/mas/src/match-media-controller.js:
```diff
@@ -0,0 +1,34 @@
+import { LitElement } from 'lit';
+
+export default class MatchMediaController {
```
is this different than https://opensource.adobe.com/spectrum-web-components/tools/match-media/ ?

can we use the lit one instead?

st-angelo-adobe replied:
We talked about it a few weeks ago. You said if `MatchMediaController` from SWC is small we could use it, otherwise we can implement the logic in js. And this is it, the implementation in js. 
However, it's pretty much the same as the one from SWC to be honest. Let me know if you want me to remove this and just use the one from SWC.

st-angelo-adobe replied:
Actually since `MatchMediaController` from SWC is so small we can just use that. I changed it back everywhere too.

yesil replied:
true true, I thought this portion was from Axel and I didn't remember our discussion.
if putting `MatchMediaController` doesn't explode our bundle size, let's use it as it.

yesil commented on libs/features/mas/src/merch-stock.js:
```diff
@@ -1,6 +1,6 @@
 import { LitElement, css, html } from 'lit';
 import { EVENT_MERCH_STOCK_CHANGE } from './constants.js';
-import { MatchMediaController } from '@spectrum-web-components/reactive-controllers/src/MatchMedia.js';
+import MatchMediaController from './match-media-controller.js';
```
I would like to understand the difference

st-angelo-adobe replied:
See above

yesil commented on libs/features/mas/src/variants/ah-try-buy-widget.js:
```diff
@@ -0,0 +1,176 @@
+      flex-direction: row;
+    }
+
+    :host([variant='ah-try-buy-widget']) .footer .spectrum-Button-label {
```
if we don't support Spectrum CSS anymore for AH, we should remove such rules.

yesil commented on libs/features/mas/src/variants/ah-try-buy-widget.js:
```diff
@@ -0,0 +1,176 @@
+  `;
+}
+
+customElements.define('ah-try-buy-widget', AHTryBuyWidget);
```
please cleanup Spectrum CSS related rules since we switched to SWC.

st-angelo-adobe replied:
I removed, hopefully all of them

3ch023 commented on libs/features/mas/src/checkout-mixin.js:
```diff
@@ -100,7 +100,6 @@ export function CheckoutMixin(Base) {
         }
 
         async render(overrides = {}) {
-            if (!this.isConnected) return false;
```
is it to fix some issue? do you have more details?

yesil replied:
@3ch023 I asked them.  Since `sp-button` doesn't have checkout-link capabilities, instead we will render a checkout-link but not in the DOM and just forward clicks to it. 
But the DOM element must be rendered even if it is not connected to DOM which is fine IMO.

3ch023 replied:
i am good if there is no impact on events we throw for CCD

3ch023 commented:
noticed some console error in mas.js 
<img width="1038" alt="image" src="https://github.com/user-attachments/assets/2e521731-db5a-470a-bf0f-6ae0a47b745d" />

3ch023 commented on libs/features/mas/src/hydrate.js:
```diff
@@ -106,38 +125,129 @@ export function processBackgroundImage(
     }
 }
 
+export function getTruncatedTextData(text, limit, withSuffix = true) {
```
could we hide extra text with styles instead of truncating description? https://developer.mozilla.org/en-US/docs/Web/CSS/text-overflow

st-angelo-adobe replied:
As mentioned yesterday in the call, we can't really do it with css only.

3ch023 commented on libs/features/mas/src/merch-card-collection.js:
```diff
@@ -1,6 +1,5 @@
 import { html, LitElement } from 'lit';
-import { MatchMediaController } from '@spectrum-web-components/reactive-controllers/src/MatchMedia.js';
-
+import MatchMediaController from './match-media-controller.js';
```
we don't use collection for AH yet, right?
trying to understand what is this change for

st-angelo-adobe replied:
I needed the `MatchMediaController`, or similar login for the try buy card. I ran it past Ilyas as well and decided to just implement the logic in a local controller. Because I did that, I changed it everywhere as well.

But the thing is, what I implemented is pretty much the same as the SWC implementation, so I just scrapped my local controller, and went back to using it from SWC. So this is no longer a thing

3ch023 reviewed (approved):
Looks good!
added a few comments to fix/discuss.
Also, could you please add a simple nala test, similar to masccd? but masah?

3ch023 reviewed (changes requested):
@Axelcureno @st-angelo-adobe there seem to be a regression
noticed a difference between
https://main--milo--adobecom.hlx.live/libs/features/mas/docs/merch-card.html
and 
https://mwpw-164093--milo--adobecom.hlx.live/libs/features/mas/docs/merch-card.html
merch-card is no longer marked as failed if checkout url is not resolved
<img width="382" alt="image" src="https://github.com/user-attachments/assets/980f40d1-85e5-49c2-983f-e5b8facb608d" />

cc @yesil

3ch023 commented:
@Axelcureno another thing i noticed is that mas.js still contains process.env?.DEBUG, which mean AH will have their webpack issue.
pls update tacocat-core-1.13.0.tgz package to the new one with your fix.

Axelcureno commented on libs/features/mas/src/hydrate.js:
```diff
@@ -61,11 +63,21 @@ export function processSize(fields, merchCard, sizeConfig) {
+        let title = fields.cardTitle;
+        const { maxCount } = titleConfig;
+        if (maxCount) {
+            const [truncatedTitle, cleanTitle] = getTruncatedTextData(fields.cardTitle, maxCount);
```
Yes, created: https://jira.corp.adobe.com/browse/MWPW-167766

Axelcureno commented:
> noticed some console error in mas.js <img alt="image" width="1038" src="https://private-user-images.githubusercontent.com/6617234/411979333-2e521731-db5a-470a-bf0f-6ae0a47b745d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mzk0MTMwMzQsIm5iZiI6MTczOTQxMjczNCwicGF0aCI6Ii82NjE3MjM0LzQxMTk3OTMzMy0yZTUyMTczMS1kYjVhLTQ3MGEtYmYwZi02YWUwYTQ3Yjc0NWQucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDIxMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTAyMTNUMDIxMjE0WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YzQzYzI5YmFiYmVkNWM4MzlhOWFmZGZjZDMzODU1OWZkMTYwMDMwMWMxYzFmN2Y4YTNkYjU5Nzk5NGJlOWRhNSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.8Gti2ZqtorKJorSwBMFZI18PdLVHZUUcas7UmFE23Ys">

this is now fixed @3ch023

Axelcureno commented:
> @Axelcureno @st-angelo-adobe there seem to be a regression noticed a difference between https://main--milo--adobecom.hlx.live/libs/features/mas/docs/merch-card.html and https://mwpw-164093--milo--adobecom.hlx.live/libs/features/mas/docs/merch-card.html merch-card is no longer marked as failed if checkout url is not resolved <img alt="image" width="382" src="https://private-user-images.githubusercontent.com/6617234/412387340-980f40d1-85e5-49c2-983f-e5b8facb608d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mzk0MTMwMzUsIm5iZiI6MTczOTQxMjczNSwicGF0aCI6Ii82NjE3MjM0LzQxMjM4NzM0MC05ODBmNDBkMS04NWU1LTQ5YzItOTgzZi1lNWI4ZmFjYjYwOGQucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDIxMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTAyMTNUMDIxMjE1WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9OGRmZWFiNDRlNDk3ZmFmNDIwZDE1MzUwZDE2NzUyYmFmZWM5OGQ5YTI1MWMwYWQ0ZGFkNmUwMDU1YzgxMTU4YiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.qWXF2ePu8wizOS5EHB0HqTEaUDgC-EedVAZEiXOwbeo">
> 
> cc @yesil

I discovered this is caused by @bozojovicic's PR that was recently merged: https://github.com/bozojovicic/milo/blob/08b47405e1659883d82d28f1144e42d81a09d418/libs/features/mas/src/merch-card.js#L379

I removed the updateComplete from Promise.race(), not sure why was added but basically this means that if the fragment finishes reloading first it returns undefined instead of false, bypassing offer resolution errors. You can check it works now (it doesn't reproduce in main because Bozo's PR is in stage only atm).

Axelcureno commented:
> @Axelcureno another thing i noticed is that mas.js still contains process.env?.DEBUG, which mean AH will have their webpack issue. pls update tacocat-core-1.13.0.tgz package to the new one with your fix.

Thanks! Updated taco, it's now removed.

3ch023 commented on libs/features/mas/src/variants/ah-try-buy-widget.js:
```diff
@@ -0,0 +1,160 @@
+  size: ['single', 'double', 'triple'],
+};
+
+export class AHTryBuyWidget extends VariantLayout {
```
i would be also with current name, might get some actual widget later to implement?

yesil replied:
I'm for using self explanatory names. `AdobeHomeTryBuyWidget` is also fine to me.

st-angelo-adobe replied:
I like `AHTryBuyWidget` because it's in line with the CCD stuff. Plus I've renamed it 3 times already 😁  . I will do it again if everyone agrees on one name, we can maybe decide it in a dev call.

yesil replied:
it is fine too. no need to change.

3ch023 reviewed (commented):
Hi @Axelcureno , approving, but either you or QA would need to add nala tests for this.
Could be a follow up PR though

yesil commented on libs/features/mas/test/hydrate.test.js:
```diff
@@ -514,3 +550,127 @@ describe('processDescription', async () => {
 });
+
+describe('getTruncatedTextData', () => {
+  it('closes any open tags in truncated text', () => {
```
can you add a test case for graceful handling in case an error happens inside `getTruncatedTextData`

yesil commented on libs/features/mas/src/hydrate.js:
```diff
@@ -106,36 +136,127 @@ export function processBackgroundImage(
+    }
+}
+
+export function getTruncatedTextData(text, limit, withSuffix = true) {
```
This method looks a bit sensible, should we try/catch and fallback to default value?

st-angelo-adobe replied:
Actually I'm finding hard to find a scenario where it fails, other than if the text is `null`/`undefined`, which I've accounted for now with a new commit.

One issue that I'm spotting though is if `<` or `>` are inside the text, as actual characters, the result isn't quite what you'd want. Doesn't break, but isn't right either.

I assume we should account for this scenario, right?

Axelcureno replied:
added try/catch

yesil commented on libs/features/mas/src/hydrate.js:
```diff
@@ -175,17 +296,31 @@ function createSpectrumSwcButton(cta, aemFragmentMapping, isOutline, variant) {
+          spectrumCta.addEventListener('click', (e) => {
             e.stopPropagation();
-            cta.click();
+            checkoutBtn.click(e);
```
`click` method does not take parameters.

https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/click

yesil commented on libs/features/mas/src/merch-card.js:
```diff
@@ -376,7 +376,7 @@ export class MerchCard extends LitElement {
             setTimeout(() => resolve(false), MERCH_CARD_LOAD_TIMEOUT),
         );
-        const success = await Promise.race([successPromise, timeoutPromise, this.aemFragment?.updateComplete]);
+        const success = await Promise.race([successPromise, timeoutPromise]);
```
@bozojovicic didn't you add `this.aemFragment?.updateComplete`? 
I think we need to keep it.

3ch023 replied:
not sure we can keep it if we want merch-card to fire an error when price/cta failed - and thats our contract for ccd at the moment

3ch023 replied:
this basically means we ignore successPromise

yesil replied:
@bozojovicic noticed in his autoblock PR, that when you call `merchCard.checkReady()` too early, it does not work (gives false positive) because you first need to wait for the fragment to be loaded.
if the fragment is not resolved, we cannot know which offers are to be loaded anyway.

Roycethan commented:
@Axelcureno 
1) When used dark theme , the theme is not apllied to few cards in double and single plan types:
https://mwpw-164093--milo--adobecom.hlx.live/libs/features/mas/docs/adobe-home.html?theme=dark
<img width="740" alt="image" src="https://github.com/user-attachments/assets/b010667c-c148-4266-b69f-246a02944fe1" />

st-angelo-adobe commented:
@Roycethan 
2, 3: Done.
4. I seriously doubt Accent buttons need to change to Primary color when hovered. I'll confirm with the AH team if this is an actual requirement, it may have been a misunderstanding.
5. Actually, the title is truncated at 37 characters, plus 3 for ellipses = 40 in total. If this is not the desired behavior we can change that.
6. From what I'm seeing this is actually fine. The editor is counting more than actual characters, possibly (maybe `\n` is counted as 2 characters, not sure, this needs to be investigated).
![image](https://github.com/user-attachments/assets/0d6b6e0b-f07c-4742-abda-c9e4480e5a8e)
8. That was just a remnant, previously `Triple` was just `Normal`, then you had the option to pick `Single` or `Double`. I changed it to `Triple`.

For 1 and 7 we'll look into it. cc: @Axelcureno

Roycethan commented:
> @Axelcureno Can you plz fix following , Thanks
> 
> 1. When used dark theme , the theme is not apllied to few cards in double and single plan types:
>    https://mwpw-164093--milo--adobecom.hlx.live/libs/features/mas/docs/adobe-home.html?theme=dark
> 
> <img alt="image" width="740" src="https://private-user-images.githubusercontent.com/104928176/414947915-b010667c-c148-4266-b69f-246a02944fe1.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDAwNjkxMDUsIm5iZiI6MTc0MDA2ODgwNSwicGF0aCI6Ii8xMDQ5MjgxNzYvNDE0OTQ3OTE1LWIwMTA2NjdjLWMxNDgtNDI2Ni1iNjlmLTI0NmEwMjk0NGZlMS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMjIwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDIyMFQxNjI2NDVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1kNmMxMzlmOGYxZDE5MDk5YTRmMTQ5OGMxZTA4YmI0OTliM2M5ZGNjMTRmMjQwMzExNzI0MjU4M2M1NDk3NzVkJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.VZ5jbm_rJD8gR_3ivhismB8WCLKW9WaF0Z3748k8u5g">
> 2)Plz update the first card in triple plan to have background - Grey
> 
> 3. On first card - 'See terms'' and Annual billed monthly' should be in separate rows and not wrapped - Better add pricing in between
> 
> <img alt="image" width="695" src="https://private-user-images.githubusercontent.com/104928176/414953565-87768e98-6991-4180-b677-9947624b7042.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDAwNjkxMDUsIm5iZiI6MTc0MDA2ODgwNSwicGF0aCI6Ii8xMDQ5MjgxNzYvNDE0OTUzNTY1LTg3NzY4ZTk4LTY5OTEtNDE4MC1iNjc3LTk5NDc2MjRiNzA0Mi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMjIwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDIyMFQxNjI2NDVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1jZWY2ZTc0OTFjNWNmZmY1YTJkOTQ3ZWQ0ZDQ1NDg0N2MxMzg0OTg1NWZlMTJmNzdhOGRmZTMwMjdmZmZkYTE3JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.XETn3i08aQbOnfGqf4F98PDPWDxc5wV9YkganNQD1E4">
> 4. ac: buttons should be responsive to hover state, e.g. S2/Accent colored button changes to S2/Primary when hovered over - I don't see hover and change color working in gallery , Can you plz check this requirement ?
> 5. ac: product titles should be truncated with ellipses (...) if they exceed 40 characters or 3 lines
>    Looks like its truncating title before 40 characters or 3 lines -
>    ex: Creative Cloud All Apps Long Test Exa...
> 
> <img alt="image" width="242" src="https://private-user-images.githubusercontent.com/104928176/414955739-d2a230d5-6de8-4645-bca7-95a1c82a7d97.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDAwNjkxMDUsIm5iZiI6MTc0MDA2ODgwNSwicGF0aCI6Ii8xMDQ5MjgxNzYvNDE0OTU1NzM5LWQyYTIzMGQ1LTZkZTgtNDY0NS1iY2E3LTk1YTFjODJhN2Q5Ny5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMjIwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDIyMFQxNjI2NDVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT04ZmJiOTI2YmQ1YTYyMzZlYWEzYzIwM2Y1ZTllY2UyYjI0ZjAyOGVlMDJhMzE1NWI3ZGI4YzYxNjlmNmJkZDI0JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.qY2gGCcuFphGRdHukNpSY3z5iWWLTXcrwmS6014cpjc">
> 6. max character count in description should be 200 characters - Its taking more than that
> 
> <img alt="image" width="856" src="https://private-user-images.githubusercontent.com/104928176/414959005-5b2e3712-e005-4832-a6b7-6eb70516bbbc.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDAwNjkxMDUsIm5iZiI6MTc0MDA2ODgwNSwicGF0aCI6Ii8xMDQ5MjgxNzYvNDE0OTU5MDA1LTViMmUzNzEyLWUwMDUtNDgzMi1hNmI3LTZlYjcwNTE2YmJiYy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMjIwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDIyMFQxNjI2NDVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT05Nzc0NjhhNTA0ZTczY2Y0Mjk0NWM0ZGUwN2M3MjIxZTE1ZTRhYjkzY2I1OWQ2NGZmOGNjOGQ5NDY4Mzk1YTQwJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.XD6DSwJav3p-ecsgamCk6LrySG2O5qlSSAfRcGsf1Hg">
> 7. In description field , if i enter around more than 25 characters in 1 row , its not wrapped into next line automatically and remaining characters are not rendered.
> 
> <img alt="image" width="1137" src="https://private-user-images.githubusercontent.com/104928176/414960926-1d51e701-52dc-441d-85b3-d4440f630d7b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDAwNjkxMDUsIm5iZiI6MTc0MDA2ODgwNSwicGF0aCI6Ii8xMDQ5MjgxNzYvNDE0OTYwOTI2LTFkNTFlNzAxLTUyZGMtNDQxZC04NWIzLWQ0NDQwZjYzMGQ3Yi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMjIwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDIyMFQxNjI2NDVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT05MTQ4ODYwNGU4NDI3MGQyYzNhMTAyYjU4MWZjYTUyMTI5MWZlMDI1YzI3ZWEyYjNmZmExMzU4MGRiMGRlMjg5JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.sTcwPu-BxpN6XL_7SzDwlAGQkpI5-s-sjq7t48htCDw">
> 8. why the 'size' feild is blank for this card ?
>    https://mwpw-164093--mas--adobecom.hlx.page/studio.html?milolibs=MWPW-164093--milo--adobecom#path=adobe-home&page=content&query=f07084e9-1963-4877-8ab0-f7d2982e62cb
> 
> <img alt="image" width="1137" src="https://private-user-images.githubusercontent.com/104928176/414962344-6c009508-3c63-479c-90ae-a66f22c06459.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDAwNjkxMDUsIm5iZiI6MTc0MDA2ODgwNSwicGF0aCI6Ii8xMDQ5MjgxNzYvNDE0OTYyMzQ0LTZjMDA5NTA4LTNjNjMtNDc5Yy05MGFlLWE2NmYyMmMwNjQ1OS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMjIwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDIyMFQxNjI2NDVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT05NjEzN2Q1ZTUwMzI3MGI4ZTM2YjMxMjZjNWU3YmFiNmI1NWViZTE5ZThhMWFiYTAwZTEyMGNmYzAzMzkyMTlkJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.Ssa9GP3G1RupGyY7wu5IqQ6j6ND0elE6ftoCeraAMiI">


@Axelcureno  cc @st-angelo-adobe 
1 - Need fix
2,3 -fixed
4- Still a Question to PDM may be since its mentioned as AC
5- Ok as explained
6 - Not working as expected
7 - Need investigation
8- fixed

Axelcureno commented:
@Roycethan we are not delivering "widgets", we are only delivering the cards. The page in milo is just for demonstration purposes, it does not represent the actual widget. The widget containing the cards is built in adobe home code.

Axelcureno commented:
@Roycethan item #9 is a mistake in the color of figma, we are not overriding colors in dark theme and even in figma it says "Dark mode follows the same S2 guidelines as Light mode."
I've left a comment here for correction: https://www.figma.com/design/7tUtNgFelfMjgPoJ5QcE1k?node-id=5160-32869&m=dev#1142597000

Axelcureno commented:
I've fixed the remaining items, please check @Roycethan

Roycethan commented:
> I've fixed the remaining items, please check @Roycethan

Thanks 1 is fixed, for 6 and 7 will create a separate followup ticket, approving this to proceed to stage

narcis-radu commented:
> Thanks 1 is fixed, for 6 and 7 will create a separate followup ticket, approving this to proceed to stage
@Roycethan , @Axelcureno , @st-angelo-adobe - Have you created follow-up tickets for 6 / 7 ?

Axelcureno commented:
> le for someone outside the Merch team to test the changes. I had to go through comments to find a URL where the elements are rendered. T

@Roycethan 6 and 7 are working as expected. I have validated on my end.

Roycethan commented:
> https://jira.corp.adobe.com/browse/MWPW-168322

Cool thanks @Axelcureno for fixing them, closing as fixed
